test/test_nn.py

```diff
@@ -2192,6 +2192,21 @@ def test_RNN_cell(self):
 
                 hx.sum().backward()
 
+    def test_RNN_cell_no_broadcasting(self):
+        def test(cell_module, hx, output_size):
+            input = Variable(torch.randn(3, 10))
+            cell = cell_module(10, hidden_size)
+            self.assertRaises(RuntimeError, lambda: cell(input, hx))
+
+        for hidden_size in [20, 1]:
+            bad_hx = Variable(torch.randn(1, hidden_size))
+            good_hx = Variable(torch.randn(3, hidden_size))
+
+            test(nn.RNNCell, bad_hx, hidden_size)
+            test(nn.GRUCell, bad_hx, hidden_size)
+            test(nn.LSTMCell, (bad_hx, good_hx), hidden_size)
+            test(nn.LSTMCell, (good_hx, bad_hx), hidden_size)
```

Comment:
The problem in here is that you always have incorrect batch size, so you're not even checking if bad hidden size fails. What I meant is testing that correct batch size and bad hidden size fails.

Reply:
Ah, sorry, I misunderstood. Yes I missed that case -- thanks for catching it!